toolchain/check/handle_require.cpp

```diff
@@ -153,9 +153,9 @@ static auto TypeStructureReferencesSelf(
 }
 
 struct ValidateRequireResult {
-  SemIR::FacetType facet_type;
-  SemIR::TypeId facet_type_type_id;
-  const SemIR::IdentifiedFacetType* identified;
+  // The TypeId of a FacetType.
+  SemIR::TypeId constraint_type_id;
```

Comment:
I feel like I'm missing something -- if this is nothing more or less than the TypeId of a FacetType, why is `constraint_type_id` a better name for it than `facet_type_type_id`?

Reply:
I found `type_type` kind of ambiguous/weird/confusing. This is the name that is used in the function below.